modules/calendar/calendar.php

```diff
@@ -489,6 +489,33 @@ function do_ics_escaping( $text ) {
 		return $text;
 	}
 
+	/**
+	 * Convert a time string into a timestamp with the proper GMT offset
+	 * @param     $time_string
+	 * @param int $offset_in_seconds
+	 *
+	 * @return string|false
+	 */
+	public static function ics_format_time( $time_string, $offset_in_seconds = 0) {
+
+		// Timestamp it
+		$timestamp = strtotime( $time_string );
+
+		if( ! $timestamp ) {
+			return false;
+		}
+
+		// Add GMT Offset
+		$timestamp += -1 * get_option('gmt_offset') * HOUR_IN_SECONDS;
+
+		// Add manual offset
+		$timestamp += $offset_in_seconds;
+
+		// \T and \Z are escaped for literal T and Z characters
+		return date( 'Ymd\THis\Z', $timestamp );
```

Comment:
This doesn't seem to be returning a timestamp like the function declaration says it is.